% -*- mode: noweb; noweb-default-code-mode: R-mode; -*-
\documentclass[11pt]{article}
%% Set my margins
\setlength{\oddsidemargin}{0.0truein}
\setlength{\evensidemargin}{0.0truein}
\setlength{\textwidth}{6.5truein}
\setlength{\topmargin}{0.0truein}
\setlength{\textheight}{9.0truein}
\setlength{\headsep}{0.0truein}
\setlength{\headheight}{0.0truein}
\setlength{\topskip}{0pt}
%% End of margins

% \usepackage{subfigure}

%%\pagestyle{myheadings}
%%\markboth{$Date$\hfil$Revision$}{\thepage}
\usepackage[pdftex,
bookmarks,
bookmarksopen,
pdfauthor={Jingsi Ming},
pdftitle={LSMM Vignette}]
{hyperref}

\title{`\texttt{LSMM}' Package1 to integrating functional
annotations with genome-wide association studies}
\author{Jingsi Ming$~^1$, Mingwei Dai$~^{1,2}$, Mingxuan Cai$~^1$, Jin Liu$~^3$, and Can Yang$~^4$\\
$~^1$ Department of Mathematics, Hong Kong Baptist University, Hong Kong.\\
$~^2$ School of Mathematics and Statistics, Xi'an Jiaotong University, Xi'an, China.\\
$~^3$ Centre of Quantitative Medicine, Duke-NUS Graduate Medical School, Singapore.\\
$~^4$ Department of Mathematics, The Hong Kong University of Science and Technology,\\
Hong Kong.\\
}

\date{\today}

\SweaveOpts{engine=R, echo=TRUE, pdf=TRUE}

\begin{document}
\SweaveOpts{concordance=TRUE}
\maketitle

\section{Overview}

This vignette provides an introduction to the `\texttt{LSMM}' package.
R package `\texttt{LSMM}' implements LSMM (Latent Sparse Mixed Model),
an efficient statistical approach to integrating functional annotations with genome-wide association studies. It provides model parameter estimation and statistical inference for risk SNPs and relevant annotations.

The package can be loaded with the command:

<<preliminaries,echo=FALSE,results=hide>>=
options(prompt = "R> ")
@

<<LSMM-prelim>>=
library("LSMM")
@

This vignette is organized as follows.
Section \ref{fitting} discusses how to fit LSMM in various settings.
Section \ref{association} explains command lines for statistical inference of risk SNPs and relevant annotations using LSMM.

Always feel free to contact Jingsi Ming at \texttt{jsming@life.hkbu.edu.hk} for any questions or suggestions regarding the `\texttt{LSMM}' package.

\section{Workflow}\label{workflow}

In this vignette, we use the simulated ExampleData in the package. We set the number of SNPs, fixed effects and random effects to be 100,000, 10 and 500 respectively. Users can find the $p$-value in the `\texttt{ExampleData\$Pvalue}', fixed effects in `\texttt{ExampleData\$Z}' and random effects in `\texttt{ExampleData\$A}'.

<<LSMMExample-prelim>>=

data(ExampleData)
Pvalue <- ExampleData$Pvalue
Z <- ExampleData$Z
A <- ExampleData$A
length(Pvalue)
dim(Z)
dim(A)
@
The length of `\texttt{Pvalue}' is assumed to be the same as the number of rows of matrix provided to `\texttt{Z}' and `\texttt{A}'. When we analyze real data, we need to take the intersection of the SNPs in `\texttt{Pvalue}', `\texttt{Z}' and `\texttt{A}' and keep the SNPs in the same order.

\subsection{Fitting the LSMM}\label{fitting}

We are now ready to fit LSMM using the data described above.
R package \texttt{LSMM} provides flexible analysis framework
and automatically adjusts its model structure based on the provided data.

First, assuming that there is no annotation data, we fit LSMM with the command:
<<LSMM-noZA-show,eval=FALSE>>=
fit.LSMM.noZA <- LSMM(Pvalue, Z = NULL, A = NULL)
@
or equivalently (which is actually simpler command),
<<LSMM-noZA-simple,eval=FALSE>>=
fit.LSMM.noZA <- LSMM(Pvalue)
@
Now, LSMM reduces to the Two Groups Model.

When we also have related annotation data, this annotation data can be easily incorporated into LSMM by providing it in the second or the third argument of `\texttt{LSMM}' function. The second argument is regarded as fixed effect and the third argument is regarded as random effect.
If we only consider fixed effects, then we can fit LSMM with the command:
<<LSMM-Z-show,eval=FALSE>>=
fit.LSMM.Z <- LSMM(Pvalue, Z = Z, A = NULL)
@
If we only consider random effects,
<<LSMM-A-show,eval=FALSE>>=
fit.LSMM.A <- LSMM(Pvalue, Z = NULL, A = A)
@
If we consider both fixed and random effects,
%<<LSMM-ZA-show,echo=TRUE, results=hide>>=
<<LSMM-ZA-show>>=
fit.LSMM.ZA <- LSMM(Pvalue, Z = Z, A = A)
@
`\texttt{fit.LSMM.ZA}' is a list containing parameter estimation, the posterior probability and iteration times of each stage and the value of lower bound of log-likelihood.
<<LSMM-show-ZA>>=
str(fit.LSMM.ZA)
@

\subsection{Statistical inference for risk SNPs and relevant annotations}\label{association}

Now, based on the fitted LSMM, we can make statistical inference for risk SNPs:
<<LSMM-assoc.SNP>>=
assoc.SNP.LSMM <- assoc.SNP(fit.LSMM.ZA, FDRset = 0.1, fdrControl="global")
str(assoc.SNP.LSMM)
table(assoc.SNP.LSMM$gamma.stage1)
table(assoc.SNP.LSMM$gamma.stage2)
table(assoc.SNP.LSMM$gamma)
@
`\texttt{assoc.SNP}' function returns list of binary values indicating association of SNPs for the phenotype under different stages, where one indicates that the SNP is associated with the phenotype and zero otherwise. `\texttt{assoc.SNP}' allows both local (`\texttt{fdrControl="local"}') and global FDR controls (`\texttt{fdrControl="global"}') and users can set the threshold using the argument `\texttt{FDRset}'. For ExampleData, Two Groups Model (stage1) detected 15277 SNPs, whereas LFM which uses fixed effects (stage2) and LSMM which integrates
both fixed effects and random effect identified 15602 and 25059 SNPs respectively, under the global FDR control at 0.1 level.

We can also make statistical inference for relevant annotations:
<<LSMM-relev.Anno>>=
relev.Anno.LSMM <- relev.Anno(fit.LSMM.ZA, FDRset = 0.1, fdrControl="local")
str(relev.Anno.LSMM)
table(relev.Anno.LSMM)
@
`\texttt{relev.Anno}' function returns a list of binary values indicating relevance of annotations for the phenotype, where one indicates that the annotation is relevant to the phenotype and zero otherwise. `\texttt{relev.Anno}' allows both local (`\texttt{fdrControl="local"}') and global FDR controls (`\texttt{fdrControl="global"}') and users can set the threshold using the argument `\texttt{FDRset}'. For ExampleData, LSMM identified 78 relevant annotations, under the local FDR control at 0.1 level.


\begin{thebibliography}{99}
Jingsi Ming, Mingwei Dai, Mingxuan Cai, Jin Liu, Can Yang,
LSMM: A statistical approach to integrating functional annotations with genome-wide association studies. 2017. Under review.

\bibitem{GPA} Chung D*, Yang C*, Li C, Gelernter J, and Zhao H (2013),
``GPA: A statistical approach to prioritizing GWAS results
by integrating pleiotropy information and annotation data.''
To appear in \textit{PLoS Genetics}. (* Joint first authors)

\end{thebibliography}

\end{document}


